### ISSUE TITLE
DynamoStore: Needs to write through tip to guarantee order on DDB streams

### ISSUE THREAD

**Issue by bartelink** [jet/equinox#403]
See #401 for details

TL;DR:
- CosmosDB ChangeFeed guarantees items/updates will be supplied to readers in order of when they are last touched for all items within a given logical partition
- DynamoDB only guarantees updates will follow earlier inserts/updates at the item level

Hence:
- In CosmosDB, new writes can go straight into a Calf document once it's determined that a calving is due (as long as the calf insert is before the tip update in the transaction batch)
- In DynamoDB, the only thing that can guarantee that the Indexer (or any Reactor) will see the events in the correct order is that they were all written to the same Item (i.e. the Tip).
